queries.sql

```diff
@@ -1,108 +1,174 @@
+select name from animals
+JOIN owners on animals.owner_id = owners.id 
+WHERE full_name = 'Melody Pond';
 
 -- List of all animals that are pokemon (their type is Pokemon).
-SELECT animals.name, species.name FROM animals JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+select animals.*, species.name as species_name from animals
+JOIN species on animals.species_id = species.id
+WHERE species.name = 'Pokemon';
 
 -- List all owners and their animals, remember to include those that don't own any animal.
-SELECT name, full_name FROM animals FULL JOIN owners ON animals.owner_id = owners.id;
+SELECT animals.name as animal_name, owners.fulL_name as owner_name from owners
+LEFT JOIN animals on owners.id = animals.owner_id;
 
 -- How many animals are there per species?
-SELECT COUNT(animals.name), species.name FROM animals JOIN species ON animals.species_id = species.id GROUP BY species.name;
+SELECT species.name, COUNT(*) from animals
+JOIN species ON species.id = animals.species_id
+GROUP BY species.name;
 
 -- List all Digimon owned by Jennifer Orwell.
-SELECT owners.full_name, animals.name AS name_of_animals, species.name AS name_of_species FROM animals JOIN owners ON owners.id = animals.owner_id JOIN species ON species.id = animals.species_id WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
+SELECT
+	animals.name as animal_name,
+	owners.full_name as owner_name,
+	species.name as species_name 
+from animals
+JOIN species
+	ON species.id = animals.species_id
+JOIN owners
+	ON owners.id = animals.owner_id
+WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
 
 -- List all animals owned by Dean Winchester that haven't tried to escape.
-SELECT name, escape_attempts, full_name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE escape_attempts = 0 AND full_name = 'Dean Winchester';
+SELECT * from animals
+JOIN owners
+	ON animals.owner_id = owners.id
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
 
 -- Who owns the most animals?
-SELECT owners.full_name, COUNT(animals.owner_id) as all_animals FROM animals JOIN owners ON animals.owner_id = owners.id GROUP BY owners.id ORDER BY all_animals DESC LIMIT 1;
\ No newline at end of file
+SELECT 
+	owners.full_name, 
+	COUNT(animals.name) as num_animals
+from owners
+LEFT JOIN animals
+	ON owners.id = animals.owner_id
+GROUP BY owners.full_name
+ORDER BY num_animals DESC
+LIMIT 1;
+
+-- join tables queries
+
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name, vets.name, visits.date_of_visit from vets
+JOIN visits ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animal_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(*) as num_animals_visited from vets
+	JOIN visits ON vets.id = visits.vet_id
+	WHERE vets.name = 'Stephanie Mendez';
```

Comment:
- [ ] This one is not selecting the different animals.  You need to use the keyword `distinct` like this:

```sql
SELECT DISTINCT animals.name
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez';
```

Reply:
ok